src/ngraph_builder.cc

```diff
@@ -1348,7 +1334,104 @@ tf::Status Builder::TranslateGraph(const std::vector<tf::TensorShape>& inputs,
             ng_batch_norm, ng::AxisVector{0, 2, 3, 1}, reshaped_shape);
       }
 
-      ng_op_map[op->name()] = ng_batch_norm;
+      SaveNgOp(ng_op_map, op->name(), ng_batch_norm);
+    }
+    // --------------
+    // FusedBatchNormGrad
+    // --------------
+    else if (op->type_string() == "FusedBatchNormGrad") {
+      TF_RETURN_IF_ERROR(ValidateInputCount(op, 5));
+
+      bool tf_is_training;
+      if (tf::GetNodeAttr(op->attrs(), "is_training", &tf_is_training) !=
+          tf::Status::OK()) {
+        NGRAPH_VLOG(3) << "is_training attribute not present, setting to true";
+        tf_is_training = true;
+      }
+
+      NGRAPH_VLOG(3) << "is_training: " << tf_is_training;
+
+      shared_ptr<ng::Node> ng_delta;
+      shared_ptr<ng::Node> ng_input;
+      shared_ptr<ng::Node> ng_scale;
+      shared_ptr<ng::Node> ng_mean;
+      shared_ptr<ng::Node> ng_variance;
+      TF_RETURN_IF_ERROR(GetInputNode(ng_op_map, op, 0, &ng_delta));
+      TF_RETURN_IF_ERROR(GetInputNode(ng_op_map, op, 1, &ng_input));
+      TF_RETURN_IF_ERROR(GetInputNode(ng_op_map, op, 2, &ng_scale));
+      TF_RETURN_IF_ERROR(GetInputNode(ng_op_map, op, 3, &ng_mean));
+      TF_RETURN_IF_ERROR(GetInputNode(ng_op_map, op, 4, &ng_variance));
+
+      std::string tf_data_format;
+      TF_RETURN_IF_ERROR(
+          tf::GetNodeAttr(op->attrs(), "data_format", &tf_data_format));
+
+      if (tf_data_format != "NHWC" && tf_data_format != "NCHW") {
+        return tf::errors::InvalidArgument(
+            "FusedBatchnormGrad data format is neither NHWC nor NCHW");
+      }
+
+      bool is_nhwc = (tf_data_format == "NHWC");
+      
+      NGRAPH_VLOG(3) << "data_format: " << tf_data_format;
+
+      float tf_epsilon;
+      if (tf::GetNodeAttr(op->attrs(), "epsilon", &tf_epsilon) !=
+          tf::Status::OK()) {
+        NGRAPH_VLOG(3) << "epsilon attribute not present, setting to zero";
+        tf_epsilon = 0.0001; 
+      }
+
+      NGRAPH_VLOG(3) << "epsilon: " << tf_epsilon;
+
+      // ToDo: We are temporarily supplying a fake value for beta here
+      // (all zero, same shape/et as scale/gamma), because Tensorflow does not give beta to us.
+      // This should work because nGraph should not actually use beta. The nGraph
+      // op may change to discard this parameter. Update this when nGraph does.
+      shared_ptr<ng::Node> ng_beta =
+          std::make_shared<ngraph::op::Constant>(ng_scale->get_element_type(),
+                           ng_scale->get_shape(),
+                           std::vector<std::string>{ng::shape_size(ng_scale->get_shape()),"0"});
+
+      if (is_nhwc) {
```

Comment:
FYI, @megawidget has a nice refactor in progress for this sort of thing at #94. Not merged yet.

Reply:
it would be nice if mine went in first, as much as i like merging conflicts :D

Reply:
Should I do a cherry-pick from PR #94 or wait it be merged into master?

Reply:
Now #94 is merged into master - so you may want to merge master to your branch and use the helper functions.

Reply:
Made the change for `nhwc` case using helper function.